v2/processor.go

```diff
@@ -82,23 +83,55 @@ func NewProcessor(receiver Receiver, handler HandlerFunc, options *ProcessorOpti
 		}
 	}
 	return &Processor{
-		receiver:          receiver,
+		receivers:         []Receiver{receiver},
 		handle:            handler,
 		options:           opts,
 		concurrencyTokens: make(chan struct{}, opts.MaxConcurrency),
 	}
 }
 
-// Start starts the processor and blocks until an error occurs or the context is canceled.
+func (p *Processor) AddReceiver(receivers ...Receiver) {
+	p.receivers = append(p.receivers, receivers...)
+}
+
+// Start starts processing on all the receivers of the processor and blocks until all processors are stopped or the context is canceled.
+// It will retry starting the processor based on the StartMaxAttempt and StartRetryDelayStrategy.
+// Returns a combined list of errors encountered during each processor start.
+func (p *Processor) Start(ctx context.Context) error {
+	wg := sync.WaitGroup{}
+	errChan := make(chan error, len(p.receivers))
+	for i := range p.receivers {
+		wg.Add(1)
+		go func(index int) {
+			defer wg.Done()
+			err := p.StartAtIndex(ctx, index)
```

Comment:
I see, StartAtIndex is only used here.
Let's keep it internal. that lets us change how we organize them